file.go

```diff
@@ -0,0 +1,289 @@
+	} else {
+		val = path.Join(val, ".local", "share")
+	}
+	return val
+}
+
+// NewLocalFile creates a JSON file storage for ipam
+func NewLocalFile(ctx context.Context, path string) Storage {
+	return &file{
+		path:       path,
+		prettyJSON: true,
+		parent:     NewMemory(ctx),
+		modTime:    nullModTime,
+		lock:       sync.RWMutex{},
+	}
+}
+
+func (f *file) clearParent(ctx context.Context) (err error) {
+	namespaces, err := f.parent.ListNamespaces(ctx)
+	if err != nil {
+		return err
+	}
+	for _, namespace := range namespaces {
+		if err = f.parent.DeleteAllPrefixes(ctx, namespace); err != nil {
+			return err
+		}
+		if namespace == defaultNamespace {
+			// skip deletion instead of replicating NewMemory behavior
+			continue
+		}
+		if err = f.parent.DeleteNamespace(ctx, namespace); err != nil {
+			return err
+		}
+	}
+	return err
+}
+
+func (f *file) refresh(ctx context.Context) error {
+	if modTime := f.getModTime(); modTime != nullModTime && modTime == f.modTime {
+		return nil
+	}
+	return f.reload(ctx)
+}
+func (f *file) getModTime() time.Time {
+	info, err := os.Stat(f.path)
+	if err != nil {
+		return nullModTime
+	}
+	return info.ModTime()
+}
+
+// see ipamer.NamespacedLoad for similar, but incomplete functionality
+func (f *file) reload(ctx context.Context) (err error) {
+	var data []byte
+	storage := make(fileJSONData)
+	if _, err = os.Stat(f.path); !errors.Is(err, fs.ErrNotExist) {
+		data, err = os.ReadFile(f.path)
+		if err != nil {
+			return fmt.Errorf("failed to read state file %q: %w", f.path, err)
+		}
+	}
+	f.modTime = f.getModTime()
+	// smallest valid piece of data is "{}"
+	if len(data) >= 2 {
+		err = json.Unmarshal(data, &storage)
+		if err != nil {
+			return fmt.Errorf("failed to parse state file %q: %w", f.path, err)
+		}
+	}
+	// TODO: improve by diffing parent storage instead of discarding and recreating?
+	if err = f.clearParent(ctx); err != nil {
+		return fmt.Errorf("failed to clear memory storage: %w", err)
+	}
+	for namespace, prefixes := range storage {
+		if err = f.parent.CreateNamespace(ctx, namespace); err != nil {
+			return err
+		}
+		for _, prefix := range prefixes {
+			if _, err = f.parent.CreatePrefix(ctx, prefix.toPrefix(), namespace); err != nil {
+				return err
+			}
+		}
+	}
+	return err
+}
+
+// see ipamer.NamespacedDump for similar, but incomplete functionality
+func (f *file) persist(ctx context.Context) (err error) {
+	storage := make(fileJSONData)
+	var (
+		prefixes map[string]prefixJSON
+		ok       bool
+		data     []byte
+	)
+
+	namespaces, err := f.parent.ListNamespaces(ctx)
+	if err != nil {
+		return err
+	}
+	for _, namespace := range namespaces {
+		if prefixes, ok = storage[namespace]; !ok {
+			prefixes = make(map[string]prefixJSON)
+			storage[namespace] = prefixes
+		}
+		ps, err := f.parent.ReadAllPrefixes(ctx, namespace)
+		if err != nil {
+			return err
+		}
+		for _, prefix := range ps {
+			prefixes[prefix.Cidr] = prefix.toPrefixJSON()
+		}
+	}
+	if f.prettyJSON {
```

Comment:
this seems to be always true, so what is the purpose then ? 
Should it be a configurable option ?

Reply:
yeah, I did not want it to be a hardcoded behavior, will leave a note here